lib/renderer/src/renderer/scene/rust/vange_rs.h

```diff
@@ -0,0 +1,82 @@
+#ifndef VANGE_RS_H
+#define VANGE_RS_H
+
+#include <cstdint>
+
+/**
+  This is a stub file for the upcoming vange-rs library bindings
+**/
+
+// TODO: agree on how we deal with structs
+struct vange_rs_camera_description {
+	float fov;
+	float aspect;
+	float near;
+	float far;
+};
+
+struct vange_rs_vector3 {
+	float x;
+	float y;
+	float z;
+};
+
+struct vange_rs_quaternion {
+	float x;
+	float y;
+	float z;
+	float w;
+};
+
+struct vange_rs_transform {
+	vange_rs_vector3 position;
+	vange_rs_quaternion rotation;
+};
+
+struct vange_rs_map_description {
+	int32_t width; // H_SIZE
+	int32_t height; // V_SIZE
+
+	// data layout: first `width` bytes are height, the rest `width` bytes are meta data
+	uint8_t** lineT;
+
+	// Material offsets in the palette
+	uint8_t* material_begin_offsets;
+	uint8_t* material_end_offsets;
+
+	// 8 for world, 16 for escave
+	int32_t material_count;
+};
+
+// TODO: or it could be left, bottom, right, top
+struct vange_rs_rect {
+	int32_t x;
+	int32_t y;
+	int32_t width;
+	int32_t height;
+};
+
+
+// TODO: pointer?
+void vange_rs_camera_create(vange_rs_camera_description desc);

```

Comment:
perhaps we can use a simpler prefix, like "rv_", which would stand for "Rusty Vangers"?

Reply:
Actually I think we should avoid using "rust" and "vange-rs" naming. As I think, we doing plugin system that allow us to add different types of renderers, "vange-rs" is one of them. But all of them share same interface. So, maybe we can call it "plugin", "external", "dynamic", etc.

Reply:
maybe dl_*

Reply:
What does "dl_" stand for?
I don't particularly care what the prefix is, honestly.
And I think "rv_" is fine, too. The whole effort is driven by the need to integrate Rusty Vangers rendering. If in the future we'll get more rendering plugins, not written in Rust, that's a very good place to be in, and I don't think "rv_" prefix is going to be a problem at that time.

Reply:
dynamic linking, same as dl library. I agree not big deal